File: test_kore_install.js
#!/usr/bin/env node

/**
 * KORE v1.1.5 - npm Installation Verification Script
 * 
 * Quick test to verify KORE is installed and working correctly in Node.js
 * Run this after: npm install kore-fileformat
 * 
 * Usage:
 *   node test_kore_install.js
 */

const assert = require('assert');

async function test_npm_installation() {
    console.log('='.repeat(60));
    console.log('📦 KORE NPM INSTALLATION TEST');
    console.log('='.repeat(60));
    console.log();
    
    // Test 1: Import
    console.log('Test 1: Requiring kore-fileformat...');
    try {
        const kore = require('kore-fileformat');
        console.log('  ✅ SUCCESS - Module loaded');
    } catch (err) {
        console.log(`  ❌ FAILED - ${err.message}`);
        return false;
    }
    console.log();
    
    // Test 2: Basic compression
    console.log('Test 2: Basic compression/decompression...');
    try {
        const kore = require('kore-fileformat');
        
        const testData = Buffer.from('hello world'.repeat(100));
        const compressed = kore.compress(testData);
        const decompressed = kore.decompress(compressed);
        
        if (Buffer.compare(decompressed, testData) === 0) {
            const ratio = (compressed.length / testData.length * 100).toFixed(1);
            console.log('  ✅ SUCCESS');
            console.log(`     Original: ${testData.length} bytes`);
            console.log(`     Compressed: ${compressed.length} bytes`);
            console.log(`     Ratio: ${ratio}%`);
        } else {
            console.log('  ❌ FAILED - Data mismatch after decompression');
            return false;
        }
    } catch (err) {
        console.log(`  ❌ FAILED - ${err.message}`);
        return false;
    }
    console.log();
    
    // Test 3: Performance (100KB data)
    console.log('Test 3: Performance test (100KB data)...');
    try {
        const kore = require('kore-fileformat');
        
        const data = Buffer.alloc(100 * 1024, 'x');
        
        const t0 = process.hrtime.bigint();
        const compressed = kore.compress(data);
        const t1 = process.hrtime.bigint();
        const decompressed = kore.decompress(compressed);
        const t2 = process.hrtime.bigint();
        
        const compressTime = Number(t1 - t0) / 1e6; // ms
        const decompressTime = Number(t2 - t1) / 1e6; // ms
        
        if (Buffer.compare(decompressed, data) === 0) {
            const compressSpeed = (data.length / (compressTime / 1000) / 1e6).toFixed(0);
            const decompressSpeed = (data.length / (decompressTime / 1000) / 1e6).toFixed(0);
            const ratio = (compressed.length / data.length * 100).toFixed(1);
            
            console.log('  ✅ SUCCESS');
            console.log(`     Compression: ${compressTime.toFixed(2)}ms (${compressSpeed} MB/s)`);
            console.log(`     Decompression: ${decompressTime.toFixed(2)}ms (${decompressSpeed} MB/s)`);
            console.log(`     Ratio: ${ratio}%`);
        } else {
            console.log('  ❌ FAILED - Data mismatch');
            return false;
        }
    } catch (err) {
        console.log(`  ❌ FAILED - ${err.message}`);
        return false;
    }
    console.log();
    
    // Test 4: Different data patterns
    console.log('Test 4: Testing different data patterns...');
    try {
        const kore = require('kore-fileformat');
        
        const patterns = {
            'Repetitive': Buffer.alloc(10000, 'A'),
            'Random': Buffer.from(Array.from({length: 5000}, (_, i) => i % 256)),
            'Mixed': Buffer.from('catdogbird'.repeat(1000)),
        };
        
        for (const [name, pattern] of Object.entries(patterns)) {
            const compressed = kore.compress(pattern);
            const decompressed = kore.decompress(compressed);
            const ratio = (compressed.length / pattern.length * 100).toFixed(1);
            
            if (Buffer.compare(decompressed, pattern) === 0) {
                console.log(`  ✅ ${name}: ${ratio}% compression`);
            } else {
                console.log(`  ❌ ${name}: Data mismatch`);
                return false;
            }
        }
    } catch (err) {
        console.log(`  ❌ FAILED - ${err.message}`);
        return false;
    }
    console.log();
    
    return true;
}

// Run tests
test_npm_installation().then(success => {
    console.log('='.repeat(60));
    if (success) {
        console.log('✅ ALL TESTS PASSED - KORE IS READY TO USE!');
        console.log('='.repeat(60));
        process.exit(0);
    } else {
        console.log('❌ TESTS FAILED - KORE INSTALLATION INCOMPLETE');
        console.log('='.repeat(60));
        process.exit(1);
    }
}).catch(err => {
    console.error(`❌ UNEXPECTED ERROR: ${err.message}`);
    process.exit(1);
});

File: test_kore_install.py
#!/usr/bin/env python3
"""
KORE v1.1.5 - Installation Verification Script

Quick test to verify KORE is installed and working correctly.
Run this after: pip install kore-fileformat

Usage:
  python test_kore_install.py
"""

import sys
import time

def test_python_installation():
    """Test Python KORE installation"""
    print("=" * 60)
    print("🐍 KORE PYTHON INSTALLATION TEST")
    print("=" * 60)
    print()
    
    # Test 1: Import
    print("Test 1: Importing kore_fileformat...")
    try:
        import kore_fileformat
        print(f"  ✅ SUCCESS - Version: {kore_fileformat.__version__}")
    except ImportError as e:
        print(f"  ❌ FAILED - {e}")
        return False
    print()
    
    # Test 2: Basic compression
    print("Test 2: Basic compression/decompression...")
    try:
        from kore_fileformat import compress, decompress
        
        test_data = b"hello world" * 100
        compressed = compress(test_data)
        decompressed = decompress(compressed)
        
        if decompressed == test_data:
            ratio = len(compressed) / len(test_data) * 100
            print(f"  ✅ SUCCESS")
            print(f"     Original: {len(test_data)} bytes")
            print(f"     Compressed: {len(compressed)} bytes")
            print(f"     Ratio: {ratio:.1f}%")
        else:
            print(f"  ❌ FAILED - Data mismatch after decompression")
            return False
    except Exception as e:
        print(f"  ❌ FAILED - {e}")
        return False
    print()
    
    # Test 3: Performance
    print("Test 3: Performance test (1MB data)...")
    try:
        from kore_fileformat import compress, decompress
        
        data = b"x" * (1024 * 1024)  # 1MB
        
        start = time.time()
        compressed = compress(data)
        compress_time = time.time() - start
        
        start = time.time()
        decompressed = decompress(compressed)
        decompress_time = time.time() - start
        
        if decompressed == data:
            compress_speed = len(data) / compress_time / 1e6
            decompress_speed = len(data) / decompress_time / 1e6
            ratio = len(compressed) / len(data) * 100
            
            print(f"  ✅ SUCCESS")
            print(f"     Compression: {compress_time*1000:.2f}ms ({compress_speed:.0f} MB/s)")
            print(f"     Decompression: {decompress_time*1000:.2f}ms ({decompress_speed:.0f} MB/s)")
            print(f"     Ratio: {ratio:.1f}%")
        else:
            print(f"  ❌ FAILED - Data mismatch")
            return False
    except Exception as e:
        print(f"  ❌ FAILED - {e}")
        return False
    print()
    
    # Test 4: Different data patterns
    print("Test 4: Testing different data patterns...")
    try:
        from kore_fileformat import compress, decompress
        
        patterns = {
            "Repetitive": b"A" * 10000,
            "Random": bytes(range(256)) * 50,
            "Mixed": (b"cat" + b"dog" + b"bird") * 1000,
        }
        
        for name, pattern in patterns.items():
            compressed = compress(pattern)
            decompressed = decompress(compressed)
            ratio = len(compressed) / len(pattern) * 100
            
            if decompressed == pattern:
                print(f"  ✅ {name}: {ratio:.1f}% compression")
            else:
                print(f"  ❌ {name}: Data mismatch")
                return False
    except Exception as e:
        print(f"  ❌ FAILED - {e}")
        return False
    print()
    
    return True

if __name__ == "__main__":
    try:
        success = test_python_installation()
        
        print("=" * 60)
        if success:
            print("✅ ALL TESTS PASSED - KORE IS READY TO USE!")
            print("=" * 60)
            sys.exit(0)
        else:
            print("❌ TESTS FAILED - KORE INSTALLATION INCOMPLETE")
            print("=" * 60)
            sys.exit(1)
    except Exception as e:
        print(f"❌ UNEXPECTED ERROR: {e}")
        sys.exit(1)
